### PR TITLE
Add Sleef Implementation for maximum Kernel for ARM

### PR DESCRIPTION
**Pull request by akote123**
The NEON Vectorized<float> implementation does not use SLEEF functions for maximum Implementation. So updated maximum function with sleef calls for better performance on graviton3.It showed good performance improvement in LLM models.
The results are taken in graviton3 machine as follows:
perf_result

This maximum kernel is used in softmax. The performance timing of softmax with default and sleef change is as below:(graviton3 machine)
softmax

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @snadampal @malfet @milpuz01

### CODE REVIEW

**Comment by snadampal** [comment]
Hi @akote123 ,  the performance improvement looks good!  what tests did you run to make sure there are no precision issues with this implementation?

**Comment by akote123** [comment]
> Hi @akote123 , the performance improvement looks good! what tests did you run to make sure there are no precision issues with this implementation?

Hi @snadampal, Thank you. For accuracy check, I have taken the output of softmax layer with and without change as numpy array and checked using np.allclose() (checked for 4 to 5 different shapes)

**Review by snadampal — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Comment by sanchitintel** [comment]
Hi @akote123 @snadampal, the CI failures that occurred 2 days ago seem legit. `ciflow/trunk` runs more CI jobs than the default case, so it looks like it exposed some new failures.

**Comment by akote123** [comment]
> Hi @akote123 @snadampal, the CI failures that occurred 2 days ago seem legit. `ciflow/trunk` runs more CI jobs than the default case, so it looks like it exposed some new failures.

@sanchitintel ,Thank you. I will check and update .

**Comment by malfet** [comment]
> @pytorchbot label "topic: not user facing"

@maajidkhann why did you use this label? Not user facing should be reserved for PR that do not have any user impact, while this one improves perf, so it should be "topic: performance" or something

**Comment by maajidkhann** [comment]
> > @pytorchbot label "topic: not user facing"
> 
> @maajidkhann why did you use this label? Not user facing should be reserved for PR that do not have any user impact, while this one improves perf, so it should be "topic: performance" or something

I see, Thanks for letting me know. I will take care of it from future PR's. Right now, sleef isn't enabled in the default CPU builds on ARM, so I kind of put the "not user facing" label thinking it wouldn't impact the default pypi aarch64 binaries performance.

Will add appropriate labels for future PR's whenever its perf related.
